### PR TITLE
pycloudstack: update framework to support vtpm

### PR DESCRIPTION
**Pull request by ruomengh**
1. vmguest.py: parameters are added to indicate whether a vTPM TD needs to be created along with TD
2. vmparam.py: Add vTPM TDVF binary path and vTPM TD binary path
3. vmm.py: update destroy method so that a VM can be destroyed but not undefined
4. virtxml.py: Add method to set vtpm fields in libvirt xml template

### CODE REVIEW

File: utils/pycloudstack/pycloudstack/vmguest.py [hunk 1]
```diff
@@ -389,6 +410,17 @@ def new_vm(self, vmtype, vmspec=VMSpec.model_base(), vm_class=VMMLibvirt,
         current_time = datetime.datetime.now().strftime("%Y-%m-%d-%H-%M-%S-%f")
         vm_name = f"{vmtype}-{user_name}-{current_time}"
 
+        # vTPM BIOS path and vTPM TD log
+        if has_vtpm is True:
+            if vtpm_path is not None:
+                vtpm_path = vtpm_path
+            else:
+                vtpm_path = VTPM_PATH
```
Comment (kenplusplus) [inline utils/pycloudstack/pycloudstack/vmguest.py:418]: could we add judgement about whether vtpm_path exist?

File: utils/pycloudstack/pycloudstack/vmm.py [hunk 2]
```diff
@@ -370,11 +402,14 @@ def is_shutoff(self):
         state, _ = dom.state()
         return state == libvirt.VIR_DOMAIN_SHUTOFF
 
-    def state(self):
+    def state(self, vtpm_dom=None):
```
Comment (kenplusplus) [inline utils/pycloudstack/pycloudstack/vmm.py:405]: I think it is not good, if you change the meaning of VMM.state, you also need change VMMQemu.state(). I think here we can create a new interface state_vtpm() for VMMLibvirt only.

Comment (ruomengh) [reply utils/pycloudstack/pycloudstack/vmm.py]: Thanks for the comments. Updated.

File: utils/pycloudstack/pycloudstack/vmguest.py (hunk 1 above)
Comment (ruomengh) [reply utils/pycloudstack/pycloudstack/vmguest.py:435]: Thanks for the comments. Updated.

**Review by kenplusplus — APPROVED** [review]
LGTM

**Pull request merged**